### PR TITLE
[Maintenance][API] Decouple resource metadata from ResetPasswordHandler

### PR DESCRIPTION
**Pull request by coldic3**
| Q               | A
| --------------- | -----
| Branch?         | 1.12
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/SyliusUserExtensionTest.php [hunk 1]
```diff
@@ -133,6 +133,24 @@ public function it_creates_a_update_user_encoder_listener_for_each_user_type():
         Assert::assertSame('_password', $shopUserListenerDefinition->getArgument(4));
     }
 
+    /** @test */
+    public function it_creates_resetting_token_parameters_for_each_user_type(): void
```
Comment (lchrusciel) [inline src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/SyliusUserExtensionTest.php:137]:
```suggestion
    public function it_creates_default_resetting_token_parameters_for_each_user_type(): void
```

Can we also write one test for setting different value?

**Review by lchrusciel — COMMENTED** [review]
we need to write upgrade file for this change

File: src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/SyliusUserExtensionTest.php (hunk 1 above)
Comment (coldic3) [reply src/Sylius/Bundle/UserBundle/Tests/DependencyInjection/SyliusUserExtensionTest.php]: done :slightly_smiling_face:

File: UPGRADE-API-1.12.md [hunk 2]
```diff
@@ -69,3 +69,5 @@ Here is how the response looks like:
 1. The  `GET` `api/v2/shop/orders/{tokenValue}/payments/{payments}/methods` and `api/v2/shop/payments/{id}/methods` endpoints have been removed and changed into collection request with 2 parameters `api/v2/shop/payment-methods?paymentId={id}&orderToken={token}`.
    Now when we do not provide parameters in response it returns all available `paymentMethods` in channel.
    Wrong parameters otherwise cause empty array `[]` in response and correct parameters return `paymentMethods` available for your `payment`.
+ 
+1. The 2nd parameter `MetadataInterface` has been removed from `src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ResetPasswordHandler` and a token TTL value must be used instead as the 3rd parameter.
```
Comment (lchrusciel) [inline UPGRADE-API-1.12.md:73]:
```suggestion
1. The 2nd parameter `MetadataInterface` has been removed from `src/Sylius/Bundle/ApiBundle/CommandHandler/Account/ResetPasswordHandler` and replaced by `Sylius\Component\User\Security\PasswordUpdaterInterface` (previously third parameter). From now on a token TTL value must be used instead as the 3rd parameter. 
```

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, Kevin! :1st_place_medal:

**Pull request merged**